### PR TITLE
[3.x] Implement loading DDS textures at run-time (reverted)

### PR DESCRIPTION
**Pull request by marcinn**
This is a 3.x backport of godotengine/godot-proposals#5748.
PR for master branch (4.x) is here: godotengine/godot/pull/69085

Adds:
* Image::load_dds_from_buffer()
* ability to load DDS textures at runtime via Image::load_from_file

The DDS loader implementation is moved from texture loader to image loader.

### CODE REVIEW

File: core/image.cpp
Comment (lawnjelly) [inline core/image.cpp:3307]: Should this be `_dds_mem_loader_func`?

**Comment by lawnjelly** [comment]
This looks great to me, would be useful from a reviewing point of view if you can confirm that the block moved between the files is unchanged so we only have to review the "glue" code.

File: core/image.cpp
Comment (marcinn) [reply core/image.cpp]: thanks, fixed

**Comment by marcinn** [comment]
> would be useful from a reviewing point of view if you can confirm that the block moved between the files is unchanged so we only have to review the "glue" code.

I only changed retvals of `ERR_FAIL_COND_V` macros (now it is similar to what bmp loader returns):

**Review by lawnjelly — APPROVED** [review]
I *think* this is fine (barring the copying right notices for static checks). It seems to copy the established techniques from the other image loaders, and the actual DDS loading is unchanged.

I've been testing this out with some DDS files, and noticed a couple of problems with the DDS loader (nothing to do with this PR):
* It can't handle incomplete mipmap chains, which do occur in the wild (I may be able to PR to fix this later as I have the fix working locally)
* It can't handle cubemaps. This would be nice to handle, but the `image->load()` system currently assumes one image for one load and might require some alternative system for loading multiple images in one go (`Image` doesn't seem to have support for layers as far as I saw).

**Comment by akien-mga** [comment]
Amended myself to fix style issues.

**Comment by marcinn** [comment]
> Amended myself to fix style issues.

Thanks. To not to forget, would you do similar changes for https://github.com/godotengine/godot/pull/69085, which is shifted to 4.x for now?

**Comment by akien-mga** [comment]
Ah right, I missed that this was backporting a change which hasn't been merged in `master` yet. I can do the same change there.

I'd probably wait for the `master` PR to be merged (can be shortly after the 4.0 release) before merging the 3.x one, for consistency.

**Review by Calinou — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by TokisanGames** [comment]
Should this be reverted? See #81126

**Comment by akien-mga** [comment]
For the time being yes.

**Comment by RockyMadio** [comment]
I am not able to insert any DDS ( converted by Converseen), the file exists in the project, but not able to display itself in any Sprite or TextureRect.
How did you do it?

**Comment by TokisanGames** [comment]
We have hundreds of DDS files, both in GD3 and GD4. What happens when you double click the DDS file in the file panel? It should load in the inspector and tell you the format, like this. What do you see?

image

I convert with intel's photoshop dds plugin, nvdia's texture tools, and the gimp.
Which exact format of DDS did you convert to? 
Can you reopen your DDS file in photoshop+dds plugin or gimp?

**Comment by RockyMadio** [comment]
> We have hundreds of DDS files, both in GD3 and GD4. What happens when you double click the DDS file in the file panel? It should load in the inspector and tell you the format, like this. What do you see?
> 
> image
> 
> I convert with intel's photoshop dds plugin, nvdia's texture tools, and the gimp. Which exact format of DDS did you convert to? Can you reopen your DDS file in photoshop+dds plugin or gimp?

yes, i could open them in gimp again. in Godot 3.5 and 3.6, it appears just as a red X

**Pull request merged**
